internal/provisioner/cluster.go

```diff
@@ -42,26 +68,24 @@ func CreateCluster(provider, s3StateStore, size string, zones []string, wait int
 		return errors.Wrapf(err, "failed to stat cluster root directory %q", clusterRootDir)
 	}
 
-	outputDir := path.Join(clusterRootDir, clusterID)
+	outputDir := path.Join(clusterRootDir, cluster.ID)
 	_, err = os.Stat(outputDir)
 	if err == nil {
 		return fmt.Errorf("encountered cluster ID collision: directory %q already exists", outputDir)
 	} else if err != nil && !os.IsNotExist(err) {
 		return errors.Wrapf(err, "failed to stat cluster directory %q", outputDir)
 	}
 
-	dns := clusterDNS(clusterID)
+	logger = logger.WithField("cluster", cluster.ID)
 
-	logger = logger.WithField("cluster", clusterID)
-
-	logger.WithField("dns", dns).Info("creating cluster")
+	logger.WithField("name", kopsMetadata.Name).Info("creating cluster")
 
 	kops, err := kops.New(s3StateStore, logger)
 	if err != nil {
 		return err
 	}
 	defer kops.Close()
-	err = kops.CreateCluster(dns, provider, kopsClusterSize, zones)
```

Comment:
Should we clean any new cluster objects in the database if we error before this command? I leaked a cluster because I forgot to have my AWS credentials exported on my first creation attempt.

Reply:
I ran into this myself. My gut says not to do anything right now -- since we're just as likely to fail partway through any other command as well, and we'll want some way to track the status of the cluster and "--resume" creating it. I think this is captured by future epics.